src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/Razor/ExtractToComponentCodeActionResolver.cs

```diff
@@ -78,13 +78,36 @@ internal sealed class ExtractToComponentCodeActionResolver(
         using var _ = StringBuilderPool.GetPooledObject(out var builder);
 
         var syntaxTree = componentDocument.GetSyntaxTree();
-        foreach (var usingDirective in GetUsingsInDocument(syntaxTree))
+
+        // Right now this includes all the usings in the original document.
+        // https://github.com/dotnet/razor/issues/11025 tracks reducing to only the required set.
+        var usingDirectives = syntaxTree.GetUsingDirectives();
+        foreach (var usingDirective in usingDirectives)
+        {
+            builder.AppendLine(usingDirective.ToFullString());
+        }
+
+        // If any using directives were added, add a newline before the extracted content.
+        if (usingDirectives.Length > 0)
         {
-            builder.AppendLine(usingDirective);
+            builder.AppendLine();
         }
 
-        var extractedText = text.GetSubTextString(new TextSpan(actionParams.Start, actionParams.End - actionParams.Start)).Trim();
-        builder.Append(extractedText);
+        var indentation = GetIndentation(actionParams.Start, text);
```

Comment:
I don't think I understand why this starts at the extract start point. Shouldn't the base indentation be calculated by starting at the 0th character of the line, and count forwards until it hits non-whitespace.

```
<div>  <span>
        Hello
    </span> [|<span>
        World
    </span>|]
</div>
```

Perhaps a pathological example, but pretty sure this would calculate the base indentation as 1, but logically it should be 4.

Also, if the extraction is all on one line, then should probably skip de-denting entirely. eg:
```
<div>
    [|<span> Hello </span>|]
</div>

Reply:
Yea, I might be a bit too naive. The dedent logic should probably be something more like

1. If all content on same line, don't dedent
2. Find the line with the left most non-whitespace character.  Calculate that indentation
3. Dedent everything by that indentation

The pathological case will still be wrong but maybe closer to right? Could also take advantage of the syntax tree and make sure paired start/ends match but that gets closer into making a static formatter for razor than I currently have the brain space for

Reply:
The other approach could be to get the actual indent string from the first line, and remove it from the start of subsequent lines, if it exists, and simply ignore lines that have a space/tab mix. Of the three, Not sure either is necessarily "right"